ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/TlsConfiguration.scala

```diff
@@ -51,24 +42,65 @@ final case class TlsConfiguration(
 
   /** If enabled and all required fields are present, it returns an SslContext suitable for server usage */
   def server: Option[SslContext] =
-    if (enabled)
-      Some(
-        GrpcSslContexts
-          .forServer(
-            keyCertChainFileOrFail,
-            keyFileOrFail,
-          )
-          .trustManager(trustCertCollectionFile.orNull)
-          .clientAuth(clientAuth)
-          .protocols(if (protocols.nonEmpty) protocols.asJava else null)
-          .build
-      )
-    else None
+    if (enabled) {
+      scala.util.Using.resources(
+        keyCertChainInputStreamOrFail,
+        keyInputStreamOrFail,
+      ) { (keyCertChain: InputStream, key: InputStream) =>
+        Some(
+          GrpcSslContexts
+            .forServer(
+              keyCertChain,
+              key,
+            )
+            .trustManager(trustCertCollectionFile.orNull)
+            .clientAuth(clientAuth)
+            .protocols(if (protocols.nonEmpty) protocols.asJava else null)
+            .build
+        )
+
+      }
+    } else None
 
   /** This is a side-effecting method. It modifies JVM TLS properties according to the TLS configuration. */
   def setJvmTlsProperties(): Unit =
     if (enabled && enableCertRevocationChecking) OcspProperties.enableOcsp()
 
+  private[tls] def keyInputStreamOrFail: InputStream = {
+    val keyFileOrFail = keyFile.getOrElse(
+      throw new IllegalArgumentException(
+        s"Unable to convert ${this.toString} to SSL Context: cannot create SSL context without keyFile."
+      )
+    )
+    prepareKeyInputStream(keyFileOrFail)
+  }
+
+  private[tls] def prepareKeyInputStream(keyFile: File): InputStream = {
+    val bytes = if (keyFile.getName.endsWith(".enc")) {
+      // TODO PBATKO: How to handle problems: url connection failure, json parsing failure, etc?
```

Comment:
It's probably easiest if you throw exceptions in case of failures (because the libraries you are using are already throwing exceptions), and use our `ResourceOwner` or scalas `Using.resources` to solve the cleanup of resources.